backend/util/util.go

```diff
@@ -161,3 +177,43 @@ func IsAuthorized(w http.ResponseWriter, r *http.Request) (bool, error) {
 	}
 	return true, nil
 }
+
+// WriteCache writes a struct to it's associated cache file for
+// a given user
+func WriteCache(username, filename string, data interface{}) {
+	// Write to JSON file
+	dirLocation := fmt.Sprintf("../data/%s", username)
+	_ = os.Mkdir(dirLocation, 0755)
+
+	fileLocation := fmt.Sprintf("../data/%s/%s.json", username, filename)
+
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = ioutil.WriteFile(fileLocation, jsonData, 0777)
+}
+
+// ServeCache serves the cached result for a given user and filename
+func ServeCache(username, filename string) (string, error) {
```

Comment:
```suggestion
// GetCache returns the cached result for a given user and filename
func GetCache(username, filename string) (string, error) {
```
Also, should we need the filename here? I mean, shouldn't the server handle filenames? I think the only information needed to access it should be `username`, nothing more.

Reply:
To get the cache at a minimum we need a username and query type. We need the username to find the folder and query to find the file, we can then append on a `.json` on the end to find the file. Changing it from taking filename to query type isn't an issue and we can do it needed but it's not too important overall